kasa/credentials.py

```diff
@@ -1,12 +1,15 @@
 """Credentials class for username / passwords."""
 
 from dataclasses import dataclass, field
-from typing import Optional
 
 
 @dataclass
 class Credentials:
     """Credentials for authentication."""
 
-    username: Optional[str] = field(default="", repr=False)
-    password: Optional[str] = field(default="", repr=False)
+    username: str = field(default="", repr=False)
+    password: str = field(default="", repr=False)
+
+    def __bool__(self):
```

Comment:
I'm not sure if we should be treating blank credentials as falsey.  Whilst I have not been able to recreate it myself some of the earlier klap implementations suggested that if a device had never been connected to the cloud then the correct credentials would be empty string username and pwd.  I don't know whether a TAPO device would also work this way but I think we should treat empty string as a valid credential and rely on it failing later if they don't work.

Reply:
Agreed on that we should avoid this magic here, but I think having username & password as non-optional is still a good idea. Do you agree on that?

Reply:
Yes agreed